test/integration/scheduler_test.go

```diff
@@ -103,50 +113,171 @@ func podScheduled(c *client.Client, podNamespace, podName string) wait.Condition
+					return node == nil
+				})
+				if err != nil {
+					t.Fatalf("Failed to observe reflected update for setting unschedulable=true: %v", err)
+				}
+			},
+			schedulable: func(t *testing.T, n *api.Node, s cache.Store, c *client.Client) {
+				n.Spec.Unschedulable = false
+				if _, err := c.Nodes().Update(n); err != nil {
+					t.Fatalf("Failed to update node with unschedulable=false: %v", err)
+				}
+				err = waitForReflection(s, nodeKey, func(node interface{}) bool {
+					return node != nil && node.(*api.Node).Spec.Unschedulable == false
+				})
+				if err != nil {
+					t.Fatalf("Failed to observe reflected update for setting unschedulable=false: %v", err)
+				}
+			},
+		},
+		// Test node.Status.Conditions=ConditionTrue/Unknown
+		{
+			unschedulable: func(t *testing.T, n *api.Node, s cache.Store, c *client.Client) {
+				n.Status = api.NodeStatus{
+					Conditions: []api.NodeCondition{badCondition},
+				}
+				if _, err = c.Nodes().UpdateStatus(n); err != nil {
+					t.Fatalf("Failed to update node with bad status condition: %v", err)
+				}
+				err = waitForReflection(s, nodeKey, func(node interface{}) bool {
+					return node != nil && node.(*api.Node).Status.Conditions[0].Status == api.ConditionUnknown
+				})
+				if err != nil {
+					t.Fatalf("Failed to observe reflected update for status condition update: %v", err)
+				}
+			},
+			schedulable: func(t *testing.T, n *api.Node, s cache.Store, c *client.Client) {
+				n.Status = api.NodeStatus{
+					Conditions: []api.NodeCondition{goodCondition},
+				}
+				if _, err = c.Nodes().UpdateStatus(n); err != nil {
+					t.Fatalf("Failed to update node with healthy status condition: %v", err)
+				}
+				waitForReflection(s, nodeKey, func(node interface{}) bool {
+					return node != nil && node.(*api.Node).Status.Conditions[0].Status == api.ConditionTrue
+				})
+				if err != nil {
+					t.Fatalf("Failed to observe reflected update for status condition update: %v", err)
+				}
+			},
+		},
 	}
 
-	err = client.Pods(api.NamespaceDefault).Delete(myPod.Name, nil)
-	if err != nil {
-		t.Errorf("Failed to delete pod: %v", err)
+	for i, mod := range nodeModifications {
+		unSchedNode, err := restClient.Nodes().Create(node)
+		if err != nil {
+			t.Fatalf("Failed to create node: %v", err)
+		}
+
+		// Apply the unschedulable modification to the node, and wait for the reflection
+		mod.unschedulable(t, unSchedNode, nodeStore, restClient)
+
+		// Create the new pod, note that this needs to happen post unschedulable
+		// modification or we have a race in the test.
+		pod := &api.Pod{
+			ObjectMeta: api.ObjectMeta{Name: "my-pod"},
+			Spec: api.PodSpec{
+				Containers: []api.Container{{Name: "container", Image: "kubernetes/pause:go"}},
+			},
+		}
+		myPod, err := restClient.Pods(api.NamespaceDefault).Create(pod)
+		if err != nil {
+			t.Fatalf("Failed to create pod: %v", err)
+		}
+
+		// There are no schedulable nodes - the pod shouldn't be scheduled.
```

Comment:
To make the test faster, you could look for an event from the scheduler saying it can't schedule the pod.